finetune/datasets/generated/dyck_loader.py

```diff
@@ -0,0 +1,171 @@
+            open_count += 1
+            # Add the close character to the to be used close character list
+            unused_close_characters_stack.append(dyck_char_pair[1])
+        else:
+            # Pop off the stack
+            close_char = unused_close_characters_stack.pop()
+            dyck_word += close_char
+            remaining_closes -= 1
+
+    return dyck_word
+
+
+class DyckLoader:
+    def __init__(
+        self,
+        dyck_character_pairs: typing.List[
+            typing.Tuple[str, str]
+        ] = DYCK_CHARACTER_PAIRS,
+        min_length_pairs: int = 2,
+        max_length_pairs: int = 20,
+        min_length_answer: int = 1,
+        max_length_answer: int = 3,
+        random_seed: typing.Optional[int] = None,
+        samples: int = 100,
+    ):
+        """Loads prompt/response data from generated dyck words.
+
+        Args:
+            dyck_character_pairs (typing.List[ typing.Tuple[str, str] ], optional): Pairs of open/close characters to use. Defaults to DYCK_CHARACTER_PAIRS.
+            min_length_pairs (int, optional): Minimum number of pairs to create the dyck word from. Defaults to 2.
+            max_length_pairs (int, optional): Maximum number of pairs to create the dyck word from. Defaults to 20.
+            min_length_answer (int, optional): Minimum length of the reference answer. Defaults to 1.
+            max_length_answer (int, optional): Maximum length of the reference answer. Defaults to 3.
+            random_seed (typing.Optional[int], optional): Seed to use for all random operations if set.
+            samples (int, optional): Number of samples to generate. Defaults to 100.
+        """
+        if min_length_answer > max_length_answer:
+            raise ValueError(
+                "min_length_answer: {min_length_answer} is greater than max_length_answer {max_length_answer}."
+            )
+
+        if min_length_pairs > max_length_pairs:
+            raise ValueError(
+                "min_length_pairs: {min_length_pairs} is greater than max_length_pairs {max_length_pairs}."
+            )
+
+        if max_length_answer > max_length_pairs:
+            raise ValueError(
+                "max_length_answer: {max_length_answer} is greater than max_length_pairs {max_length_pairs}."
+            )
+
+        self.buffer: typing.List[typing.Tuple[str, str]] = []
+
+        if random_seed is not None:
+            random.seed(random_seed)
+
+        for _ in range(samples):
+            # Decide the length the reference should be.
+            reference_length = random.randint(min_length_answer, max_length_answer)
+            # We need at least as many pairs as the reference length or the min length.
+            adjusted_min_length_pairs = max(reference_length, min_length_pairs)
+            # Decide total number of pairs.
+            pair_count = random.randint(adjusted_min_length_pairs, max_length_pairs)
+
+            # Generate a dyck of the minimum pair length.
```

Comment:
nit: comment doesn't align with the code

Reply:
Fixed. thanks!